packages/components/src/components/typography/Paragraph/Paragraph.tsx

```diff
@@ -1,72 +1,16 @@
-import { ReactNode, HTMLAttributes } from 'react';
-import styled, { css } from 'styled-components';
-import { variables } from '../../../config';
+import React from 'react';
+import styled from 'styled-components';
+import { typography, TypographyStyle } from '@trezor/theme';
 
-export type ParagraphSize = 'normal' | 'small' | 'tiny';
-
-const getLineHeight = (size: PProps['size']) => {
-    switch (size) {
-        case 'small':
-            return '18px';
-        case 'tiny':
-            return 'normal';
-        default:
-            return '22px';
-    }
-};
-
-const getWeight = (size: PProps['weight']) => {
-    switch (size) {
-        case 'normal':
-            return variables.FONT_WEIGHT.REGULAR;
-        case 'medium':
-            return variables.FONT_WEIGHT.MEDIUM;
-        case 'bold':
-            return variables.FONT_WEIGHT.DEMI_BOLD;
-        default:
-            return variables.FONT_WEIGHT.REGULAR;
-    }
+export type PProps = {
+    type?: TypographyStyle;
+    children: React.ReactNode;
 };
 
-const P_SIZES: { [key: string]: string } = {
-    normal: variables.FONT_SIZE.NORMAL,
-    small: variables.FONT_SIZE.SMALL,
-    tiny: variables.FONT_SIZE.TINY,
-};
-
-const Paragraph = styled.div<PProps>`
-    font-size: ${props => P_SIZES[props.size || 'normal']};
-    line-height: ${props => getLineHeight(props.size)};
-    color: ${({ size, theme }) => (size === 'tiny' ? theme.TYPE_LIGHT_GREY : theme.TYPE_DARK_GREY)};
-    padding: 0;
-    font-weight: ${({ weight }) => getWeight(weight)};
-    ${props =>
-        props.textAlign &&
-        css`
-            text-align: ${props.textAlign};
-        `}
+const Paragraph = styled.div<{ type: TypographyStyle }>`
```

Comment:
```suggestion
const Paragraph = styled.p<{ type: TypographyStyle }>`
```

Reply:
Perhaps makes sense to make a prop to render `Paragraph` as `span`? I never liked the `P` naming as well, now when I think of it, it's tied to html and it should not be – `Paragraph` is better IMO. `<p>` can be the simply default element for wrapping the text, with an option to make it something else. Thoughts, @komret  @matejkriz ?

Reply:
Agree that `Paragraph` is better than `P`. Even for searchability in the code this is much better. There can be some use cases when we want to render it as a span, but in this case `<div>` is closer to `<p>` than `<span>`. We can add new prop for it, something like:

```jsx
<Paragraph type="callout" as="span">...</Paragraph>
```

Reply:
While it makes more sense semantically, it shouldn't be a `p` tag IMO because then you run into console warnings of "`div` cannot be child of `p`" when you use a `Tooltip` in the text. `span` applies `display: inline` by default and we don't usually want that, so I'd keep it as a `div`. If we keep it as is, we don't need the option to change element type.

Reply:
Yes, agree somewhat regarding tooltips. I am not the biggest fan of text in `divs` but it's true we have those messages popping up from time to time... 😪  Still, for styling text within a paragraph `as="span"` would be beneficial precisely for `display: inline`